samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -399,6 +415,18 @@ private void onShutDown() {
     }
   }
 
+  private void stopDrainMonitorAndCleanup() {
+    // Stop the Drain Monitor.
+    // Also, garbage collect all DrainNotifications from Drain metadata-store of the job if the following conditions
+    // are met:
+    // 1) Job is already draining
+    // 2) All containers shutdown successfully (due to drain)
+    // 3) There was no exception in the coordinator
+    drainMonitor.stop(drainMonitor.isDraining()
```

Comment:
Let's add a method in the DrainWriter/Util class to do cleanup(). Basically it will check the drain message and delete it if needed.

Reply:
Done.